CHANGELOG.md

```diff
@@ -1,4 +1,5 @@
 ### 2.3.1 (Next)
+* [#104](https://github.com/acm19/aws-request-signing-apache-interceptor/pull/104): Make request interceptors simpler and other improvements - [@acm19](https://github.com/acm19).
```

Comment:
Let's change the version to 2.4.0 since this is pretty substantial, and change this line to "Added support for `CloseableHttpAsyncClient`"? And if you feel fancy add "Supports synchronous and asynchronous requests." into README?

Reply:
Hmm, the change doesn't add any functionality, though. It just refactors the code, that's why I didn't bump the minor. About the description change I didn't add async support, I barely added the sample to test it once we add it. I explain a bit my view on how we should add it here: https://github.com/acm19/aws-request-signing-apache-interceptor/issues/101#issuecomment-1556150389

Reply:
Resolving and merging.